lib/runtime/procs/read.js

```diff
@@ -126,6 +126,12 @@ class Read extends source {
             this.logger.debug('scheduling first live read at', firstRead.toJSON());
             this.program.scheduler.schedule(firstRead.unixms(), _.bind(this.run, this));
         }
+
+        if (this.params.optimization_info.reduce_every && this.to.isEnd()) {
```

Comment:
Previously, we left it totally up to the adapter to put what it wanted into the `optimization_info` -- this would be a departure from that design since read is now poking into the contents.

What if instead we changed the contract so that in the case of optimized `reduce -every`, the adapter returns the `every` interval as part of the `defaultTimeOptions()`. Then we would change the logic at L108 above so that it sets the initial `readTo` to be `now - lag` quantized on `every` (which seems like sensible behavior anyway) which seems to me like it would accomplish the same goal but without needing to reach into the optimization info.

Somewhat related, I think it would also be sensible to disallow the combination of `read -every :x: | reduce -every :y:` in the case where the reduce is optimized.